### DISCUSSION TITLE
Will the non-availability of cache dir fail softly?

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#2291] [General]
We're evaluating setup cache on a NFS server to distribute it across mounts.

**If this NFS mount used by JuiceFS `cache-dir` becomes unavailable:**

1. JuiceFS mounts continue to serve files but directly from object storage until `cache-dir` can be reached again (**soft fail**)
2. JuiceFS mount fail instantly (**hard fail**)?

Maybe document this too.

Thanks.

**Comment by davies** [comment]
It's not recommended to use a shared storage as caching directory. 

When it fails, JuiceFS will hang too.
